### ISSUE TITLE
mips64el: TypeError: Cannot read property 'length' of undefined

### ISSUE THREAD

**Issue by paul99** [v8mips/v8mips#124]
This bug, originally from issue https://github.com/paul99/v8m-rb/issues/113 now appears in mips64 builds. But the circumstances seem different.

It shows up when built with snapshot, and it also shows up without using the `--always-opt` flag. (this was never true in any observation in bug 113.)

Here is a log of mips64el builds on the mips buildbot: https://v8.mips.com/bbv8/builders/Target%20-%20MIPS64%20-%20edge?numbuilds=200

See build 819, as an example: https://v8.mips.com/bbv8/builders/Target%20-%20MIPS64%20-%20edge/builds/819 -- fails unicodelctest & unicodelctest-no-optimization with the TypeError, and fails another unrelated test, all as flakes.

Note that build 786 is the build _just before_ commit `e5a082f0 Reland "Only use FreeSpace objects in the free list"` was landed. And it fails unicode-case-overoptimization for this same error. The are many others, before and after the commit that fixed the mips32 cases.

Failures go back to Dec 14, the oldest commit in this log.

I'll assign to Dusan for now, as Pri 2. If you can think of additional testing or characterization, we can try to find someone to do those tests for us.

**Comment by dusmil** [comment]
I have reverted optimization which enables us to use 48-bit addresses in the code, and I still see random failures.  This only means the optimization is working well and it is not the cause of the problems.

**Comment by dusmil** [comment]
The JS debugger tests fail in the same manner.
It is indicative the failures are caused by the same bug, therefore I will not put JS debugger failures descriptions in separate new issue.

```
=== mjsunit/debug-liveedit-compile-error ===
============ Stress 1/5 ============
============ Stress 2/5 ============
============ Stress 3/5 ============
============ Stress 4/5 ============
============ Stress 5/5 ============
/mnt/mnt-mcs04/new2015/gccheck/v8/test/mjsunit/debug-liveedit-compile-error.js:56: TypeError: Cannot read property 'syntaxErrorMessage' of undefined
    caught_exception.details.syntaxErrorMessage);
                            ^
TypeError: Cannot read property 'syntaxErrorMessage' of undefined
    at /mnt/mnt-mcs04/new2015/gccheck/v8/test/mjsunit/debug-liveedit-compile-error.js:56:29
Command: /mnt/mnt-mcs04/new2015/gccheck/v8/out/mips64el.release/d8 --test --random-seed=-385544925 --stress-opt --always-opt --nohard-abort --nodead-code-elimination --nofold-constants --expose-debug-as debug /mnt/mnt-mcs04/new2015/gccheck/v8/test/mjsunit/mjsunit.js /mnt/mnt-mcs04/new2015/gccheck/v8/test/mjsunit/debug-liveedit-compile-error.js
```

The Google skipped some JS debugger tests under GC stress conditions and opened issue for this.
https://codereview.chromium.org/1016063002

Sadly, the issue is no longer visible for us because they made it private.

**Comment by dusmil** [comment]
There is a good news! 
The regexp random failures are no longer reproducible after commit c67cb28 that deals with raw pointers updating in regexp code has been landed. I was expecting so much of this commit, and finally it was landed yesterday.

Tested at a54f22d41.

Checking JS debugger random failures now.

**Comment by dusmil** [comment]
It appears I do not see JS debugger random failures anymore.

Only one failure present which is strictly deterministic.

```
=== mjsunit/debug-break-inline ===
============ Stress 1/5 ============
============ Stress 2/5 ============
============ Stress 3/5 ============
============ Stress 4/5 ============
============ Stress 5/5 ============
[couldn't find pc offset for node=14]
[method: g]
[source:
function g() {\x0a  count++;\x0a  h();\x0a  var b = 1;  // Break point is set here.\x0a}
]


#
# Fatal error in , line 0
# unable to find pc offset during deoptimization
#

==== C stack trace ===============================
```

**Comment by dusmil** [comment]
The another false call to celebration, the mips64 buildbot clearly shows failures.

https://v8.mips.com/bbv8/builders/Target%20-%20MIPS64%20-%20edge

**Comment by paul99** [comment]
In 2015-04-27 v8-call, @dusmil mentioned that msjunit/extra-arguments showed this failure. Yesterday, I ran that test many times on erpro-8 board, mipsel, mips64el, both with snapshots and without, and did not have a single failure. A few combos we run 50+ times.

Does this still fail for you, and if so, what specific platform?

I am not finding many instances on the BB, but there are a few. Here is mips64el, snapshot=on, run on error-8:
https://v8.mips.com/bbv8/builders/Target%20-%20MIPS64%20-%20edge/builds/1188/steps/Check/logs/debug-sourceinfo
same test fail in build 1182

Here is extra-arguments: https://v8.mips.com/bbv8/builders/Target%20-%20MIPS64%20-%20edge/builds/1175/steps/Check/logs/extra-arguments

This build has several tests that fail with the type error - maybe it makes sense to grab this revision, and test it: https://v8.mips.com/bbv8/builders/Target%20-%20MIPS64%20-%20edge/builds/1170
(also builds 1169, 1168, likely more)

**Comment by paul99** [comment]
I missed several of these in very recent mips64 builds:

https://v8.mips.com/bbv8/builders/Target%20-%20MIPS64%20-%20edge/builds/1203/steps/Check/logs/debug-scripts-request

https://v8.mips.com/bbv8/builders/Target%20-%20MIPS64%20-%20edge/builds/1202/steps/Check/logs/debug-sourceinfo

https://v8.mips.com/bbv8/builders/Target%20-%20MIPS64%20-%20edge/builds/1197/steps/Check/logs/regress-debug-deopt-w..

https://v8.mips.com/bbv8/builders/Target%20-%20MIPS64%20-%20edge/builds/1196/steps/Check/logs/mirror-object

https://v8.mips.com/bbv8/builders/Target%20-%20MIPS64%20-%20edge/builds/1193/steps/Check/logs/for-in-loop-initializ..

**Comment by dusmil** [comment]
Finally fixed in:
https://codereview.chromium.org/1220443002/

This also fixes dromeo DOM test failure in chromium browser on mips32 (DBUG-1790)

**Comment by djpesic** [comment]
Fix commit id: 448ec36491a33111642470ad8b78e99ce3cf1975. Closing.
